flow/SystemMonitor.cpp

```diff
@@ -55,6 +57,61 @@ SystemStatistics getSystemStatistics() {
 	    machineState.folder.present() ? machineState.folder.get() : "", &ipAddr, &statState.systemState, false);
 }
 
+namespace {
+
+std::string capitalizeCgroupKey(const std::string& key) {
+	bool wordStart = true;
+	std::string result;
+	result.reserve(key.size());
+
+	for (const char ch : key) {
+		if (std::isalnum(ch)) {
+			if (wordStart) {
+				result.push_back(std::toupper(ch));
+				wordStart = false;
+			} else {
+				result.push_back(ch);
+			}
+		} else {
+			// Skip non-alnum characters
+			wordStart = true;
+		}
+	}
+
+	return result;
+}
+
+// Collects the /sys/fs/cgroup/cpu,cpuacct/cpu.stat information
+// For more information about cpu,cpuacct, check manpages for cgroup
+void reportCpuCpuAcctCpuStat(BaseTraceEvent& traceEvent) {
+	// Default path to the cpu,cpuacct
+	// See manpages for cgroup
+	static const std::string PATH_TO_CPU_CPUACCT = "/sys/fs/cgroup/cpu,cpuacct/cpu.stat";
+
+	// readFileBytes is not usable since the file is not seekable
+	std::ifstream ifs(PATH_TO_CPU_CPUACCT);
+	if (!ifs.is_open()) {
+		traceEvent.detail("NoCpuStatFile", true);
+		return;
+	}
+
+	std::string line;
+	while (std::getline(ifs, line)) {
```

Comment:
We have getMemoryInfo() to pass the memory info in similar format: https://github.com/apple/foundationdb/blob/858d1eb9fd788febf0262a6177264200e74d2325/flow/Platform.actor.cpp#L407-L425 

Maybe we can reuse it and unify the method?